src/macros.jl

```diff
@@ -237,7 +237,128 @@ macro grad(expr)
         end
     end |> esc
 end
+
+"""
+    _make_fwd_args(func, arg_list)
+
+Function `_make_fwd_args` accepts a function name and an argument
+list, returns a tuple of argument lists whose elements are: 1. the
+`arg_list` untouched, 2. a new argument list with the function as its
+first element and other elements in `arg_list` followed. E.g.:
+
+_make_fwd_args(:f, [:(a::String), :(b::TrackedReal)])
+
+returns
+
+[:(a::String), :(b::TrackedReal)], [:f, :(a::String), :(b::TrackedReal)]
+
+It also deals with varargs and variable keyword arguments, and ensures
+that at least one of the argument is tracked.
+
+"""
+function _make_fwd_args(func, xs_l)
+    has_tracked_data = any(xs_l) do arg
+        isa(arg, Expr) && arg.head == :(::) &&
+            arg.args[end] in (:(ReverseDiff.TrackedReal), :(ReverseDiff.TrackedArray),
+                              :(ReverseDiff.TrackedVector), :(ReverseDiff.TrackedMatrix),
+                              :(ReverseDiff.TrackedVecOrMat))
+    end
+
+    has_tracked_data || error("The rule should have at least one tracked argument.")
+
+    xs_r = copy(xs_l)
+    if isa(xs_r[1], Expr) && xs_r[1].head == :parameters # has kw args
+        insert!(xs_r, 2, func)
+    else
+        insert!(xs_r, 1, func)
+    end
+    return xs_l, xs_r
+end
+
+"""
+    ReverseDiff.@grad_from_chainrules Base.sin(x::TrackedReal)
+
+The `@grad_from_chainrules` macro provides a way to import
+adjoints(rrule) defined in ChainRules to ReverseDiff. One must provide
+a method signature to import the corresponding `rrule`. In the
+provided method signature, one should replace the types of arguments
+to which one wants to take derivatives with respect with
+`ReverseDiff.TrackedReal` and `ReverseDiff.TrackedArray`
+respectively. For example, we can import `rrule` of `f(x::Real,
+y::Array)` like below:
+
+
+```
+ReverseDiff.@grad_from_chainrules f(x::TrackedReal, y::TrackedArray)
+ReverseDiff.@grad_from_chainrules f(x::TrackedReal, y::Array)
+ReverseDiff.@grad_from_chainrules f(x::Real, y::TrackedArray)
+```
+
+"""
+macro grad_from_chainrules(fcall)
+    @gensym tp output_value output back closure cls_args cls_kwargs
+
+    fcall.head == :call || error("The rule should be in format of a function call.")
+    @capture(fcall, f_(xs__)) # extract information into f and xs
+    xs_l, xs_r = _make_fwd_args(f, xs)
+
+    return quote
+        $f($(xs_l...)) = $ReverseDiff.track($(xs_r...))
+        if !hasmethod($ReverseDiff.track, Tuple{typeof($f)})
+            function $ReverseDiff.track(::typeof($f), args...; kwargs...)
+                $tp = $ReverseDiff.tape(args...)
+                $output_value, $back = ChainRules.rrule($f, map(ReverseDiff.value, args)...; kwargs...)
```

Comment:
This assumes that users load ChainRules. As suggested above, we should interpolate ChainRulesCore (similar to `$ReverseDiff`) or just use `rrule` and not escape the whole expression.